### PR TITLE
Closes #2506 - fix deleting DELETED history events by the cleanup job

### LINKED ISSUE #2506: HistoryEvents of type DELETED don't get deleted by the cleanup job

### PR DESCRIPTION
**Pull request by ryzheboka**
https://sonarcloud.io/summary/new_code?id=ryzheboka_taskana&branch=TSK-2506

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → (Closes) #&lt;Issue Number&gt;: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java [hunk 1]
```diff
@@ -93,22 +88,16 @@ void should_NotCleanHistoryEventsUntilDate_When_MinimumAgeNotReached() throws Ex
...
       Instant created,
       String parentBusinessProcessId) {
     TaskHistoryEvent historyEvent = new TaskHistoryEvent();
     historyEvent.setId(IdGenerator.generateWithPrefix(IdGenerator.ID_PREFIX_TASK_HISTORY_EVENT));
     historyEvent.setUserId(userid);
-    historyEvent.setDetails(details);
-    historyEvent.setWorkbasketKey(workbasketKey);
     historyEvent.setTaskId(taskId);
     historyEvent.setEventType(type);
-    historyEvent.setOldValue(previousWorkbasketId);
```
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:111]: Why should previous WorkbasketId and WorkbasketKey be relevant here? I will include details :)

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ void should_DeleteTaskComment_For_TaskCommentId() throws Exception {
     taskService.deleteTaskComment(comment1.getId());
 
     List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
-    assertThat(taskCommentsAfterDeletion).hasSize(0);
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:86]: Why do you change this? The task comment has nothing to do with the DELETED history Event

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (holgerhagen) [reply lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:111]: But isn't WorkbasketKey at least a required field? Obviously not, but I'm surprised....

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java [hunk 3]
```diff
@@ -140,49 +132,66 @@ void should_CleanHistoryEventsUntilDate_When_SameParentBusinessTrueAndEventsQual
...
+          createTaskHistoryEvent(
+              
+              "taskId3",
+              TaskHistoryEventType.CREATED.getName(),
+              "someUserId",
+              "someDetails",
+              Instant.now().minus(20, ChronoUnit.DAYS),
+              "sameParentId");
+      historyService.create(eventToBeCleaned5);
+      TaskHistoryEvent eventToBeCleaned6 =
```
Comment (holgerhagen) [inline lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:186]: I think this testcase is correct. But I believe the more important test would be to test if a task history is cleaned if there is a DELETED event only and no COMPLETED, TERMINATED or CANCELLED event. Isn't this selection the bug/change?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:111]: Details are implemented in a way that they always contain the WorkbasketKey and other necessary information. That's probably why WorkbasketKey isn't required

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java (hunk 3 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:186]: The COMPLETED and CANCELLED events are for different TaskIds. They should not affect the events with other TaskIds. There is no TERMINATED event in this test. There are no other Endstate events other than  DELETED for the Task with the id "taskId3". Would the test case with only the DELETED event still be necessary?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java (hunk 3 above)
Comment (ryzheboka) [reply lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:186]: Yes, I will add a test case with a unique parent id

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**